gcloud-java-core/src/main/java/com/google/cloud/BatchResult.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * This class holds a single result of a batch call. {@code T} is the type of the result and {@code
+ * E} is the type of the service-dependent exception thrown when a processing error occurs.
+ */
+public abstract class BatchResult<T, E extends BaseServiceException> {
+
+  private T result;
+  private boolean completed = false;
+  private E error;
+  private List<Callback<T, E>> toBeNotified = new LinkedList<>();
+
+  /**
+   * Returns {@code true} if the batch has been completed and the result is available; {@code false}
+   * otherwise.
+   */
+  public boolean completed() {
+    return completed;
+  }
+
+  /**
+   * Returns the result of this call.
+   *
+   * @throws IllegalStateException if the batch has not been completed yet
+   * @throws E if an error occurred when processing this request
```

Comment:
I am not sure if "this request" is clear enough as oppose to "the batch request"